### PR TITLE
3791 login

### PR DESCRIPTION
**Pull request by emitche**
Adds user flow for login and logout

### CODE REVIEW

File: www/js/app.js [hunk 1]
```diff
@@ -1,11 +1,8 @@
...
+// rCredits Register
 
-// angular.module is a global place for creating, registering and retrieving Angular modules
-// 'starter' is the name of this angular module example (also set in a <body> attribute in index.html)
-// the 2nd parameter is an array of 'requires'
-// 'starter.controllers' is found in controllers.js
-var app = angular.module('starter', ['ionic', 'starter.controllers'])
+var app = angular.module('starter', ['ionic', 'controllers', 'routes'])
 
+// Can this be moved into a separate module?
```
Comment (smoyte) [inline www/js/app.js]: This is fine here for now I think.

File: www/js/controllers.js [hunk 2]
```diff
@@ -1,56 +1 @@
...
-    { title: 'Dubstep', id: 3 },
-    { title: 'Indie', id: 4 },
-    { title: 'Rap', id: 5 },
-    { title: 'Cowbell', id: 6 }
-  ];
-})
-
-.controller('PlaylistCtrl', function($scope, $stateParams) {
-});
+angular.module('controllers', []);
```
Comment (smoyte) [inline www/js/controllers.js]: I think this file can be deleted.

File: www/js/controllers/login-controller.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
+app.controller('UserCtrl', function($scope, $state, $ionicLoading, $ionicPopup) {
+  var loggedIn = false;
+
+  var messages = {
+    "loginSuccess": "You successfully logged in.",
+    "loginFailure": "You were not logged in.",
+    "logoutSuccess": "You were successfully logged out.",
+    "logoutFailure": "Logout was unsuccessful. Please try again."
+  }
```
Comment (smoyte) [inline www/js/controllers/login-controller.js]: Nice job to put this up here. We will move it to a central place later.

File: www/js/controllers/login-controller.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+  $scope.redirectToLogin = function(){
+    $state.go("app.rlogin");
+  };
+
+  // Alert
+
+  $scope.showAlert = function(message) {
+    $ionicLoading.hide();
+    var alertPopup = $ionicPopup.alert({
```
Comment (smoyte) [inline www/js/controllers/login-controller.js]: No need to assign this to a variable I think.

File: www/js/routes.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    views: {
+      'menuContent': {
+        templateUrl: 'templates/rlogin.html'
+      }
+    }
+  });
+
+  // if none of the above states are matched, use this as the fallback
+  $urlRouterProvider.otherwise('/app/rlogin');
+});
```
Comment (smoyte) [inline www/js/routes.js:33]: Looks good.

File: www/js/controllers/login-controller.js (hunk 4 above)
Comment (emitche) [inline www/js/controllers/login-controller.js]: Agreed.

**Pull request merged**